staging/src/k8s.io/cloud-provider/controllers/node/node_controller.go

```diff
@@ -483,11 +485,14 @@ func (cnc *CloudNodeController) getNodeModifiersFromCloudProvider(
 		}
 	}
 
-	// If user provided an IP address, ensure that IP address is found
-	// in the cloud provider before removing the taint on the node
-	if nodeIP, ok := ensureNodeProvidedIPExists(node, instanceMeta.NodeAddresses); ok && nodeIP == nil {
+	// If user provided a node IP address, ensure that IP address is found
+	// in the cloud provider and place it first in the node addresses list
+	// before removing the taint on the node
+	nodeAddresses, ok := preferProvidedNodeIP(node, instanceMeta.NodeAddresses)
+	if !ok {
```

Comment:
Given that these logs might be viewed after the node object has gone away or changed, it would make debugging easier if we also returned the node IP that was not present in the list of node addresses, and printed it in the "failed to find kubelet node IP..." error message.

Reply:
Addressed in latest patch, thanks.